nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,1825 @@
+      "source": [
+        "We will leverage the few-shot capabilities of large LMs to generate synthetic but very realistic samples from a mixture of real samples. Specifically, we select two real samples from our dataset and embed these samples in a carefully designed prompt. Then, the LM takes as input the prompt and generates an augmented mixed sample influenced by the sample sentences.\n",
+        "\n",
+        "\n",
+        "Generally, a prompt looks like this:\n",
+        "\n",
+        "    Each item in the following list contains a <text type> and the\n",
+        "    respective <label type>. <label type> is one of ’<label token 1>’,\n",
+        "    ..., or ’<label token N>’. \n",
+        "    <text type>: <example text 1> (<label type>: <example label 1>)\n",
+        "    ...\n",
+        "    <text type>: <example text k> (<label type>: <example label k>)\n",
+        "    <text type>:\n",
+        "\n",
+        "In our case the prompt looks like this:\n",
+        "\n",
+        "    Each item in the following list contains a tweet and the\n",
+        "    respective sentiment. Sentiment is one of ’joy’, 'surprise' or 'anger'. \n",
+        "    Tweet: i feel angered and firey (Sentiment: anger)\n",
+        "    Tweet: im feeling very peaceful about our wedding again now after having (Sentiment: joy)\n",
+        "    Tweet:\n",
+        "\n",
+        "You can find more information on text augmentation using large LMs in [GPT3Mix](https://arxiv.org/abs/2104.08826) paper."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "qj2Of7vGILFJ"
+      },
+      "source": [
+        "First, we should extract pairs of samples from the train set. There are various extraction strategies that can be used to increase the quality of the synthetic samples. We will simply extract the pairs randomly since by repeating random sampling a diverse synthetic dataset will be created."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "zlI7GYwbGx5m"
+      },
+      "source": [
+        "def get_two_random_samples():\n",
+        "    # define a function that returns two random samples from the train set.\n",
+        "    s1, s2 = random.sample(range(0, len(emotion_train_ds)), 2)\n",
+        "    return emotion_train_ds['text'][s1], emotion_train_ds['label'][s1], emotion_train_ds['text'][s2], emotion_train_ds['label'][s2]\n",
+        "\n",
+        "def get_prompt(text1, label1, text2, label2):\n",
+        "    # define a function that takes as input two samples and generates the prompt\n",
+        "    # that we should pass to the GPT-3 language model for completion.\n",
+        "    description = \"Each item in the following list contains a tweet and the respective sentiment. Sentiment is one of 'joy', 'surprise' or 'anger'.\"\n",
+        "    prompt = (f\"{description}\\n\"\n",
+        "            f\"Tweet: {text1} (Sentiment: {idx2label[label1]})\\n\"\n",
```

Comment:
when I tried to run the notebook, it couldn't find these two functions anymore, probably due to the refactoring with the mapping > to adapt